xds/README.md

```diff
@@ -7,115 +7,153 @@ upstream endpoints and filter configuration from some external source, and then
-      any connected Quilkin proxies.
-
-      > Since an Agones GameServer can have multiple ports exposed, if multiple ports are in
-      > use, the server looks for the port named `default` and picks that as the endpoint's
-      > port (otherwise it picks the first port in the port list).
-
-   1. Filter chain is configurable on a per-proxy basis. By default an empty filter chain is
-      used and from there the filter chain can configured using annotations on the proxy's pod.
-      The following annotations are currently supported:
-      - **quilkin.dev/debug-packets**: If set to the value `true`, then a `Debug` filter will be
-        added to the filter chain, causing all packets will be logged.
-      - **quilkin.dev/routing-token-suffix-size**: Sets the size (in number of bytes) of routing tokens appended to
-        packets. Extracted tokens will matched against available endpoints in order to figure out
-        where to send the associated packet.
-        Note that the token is stripped off the packet. This annotation cannot be provided together with
-        `quilkin.dev/routing-token-prefix-size`.
-      - **quilkin.dev/routing-token-prefix-size**: Works exactly the same as `quilkin.dev/routing-token-suffix-size`
-        with the difference that the token is a prefix on the packet rather than a suffix.
-   
-   As an example, the following runs the server against a cluster (using default kubeconfig configuration) where Quilkin pods run in the `quilkin` namespace and game-server pods run in the `gameservers` namespace:
-
-   ```sh
-   go run controller.go -- port=18000 --proxy-namespace=quilkin --game-server-namespace=gameservers
-   ```
-
-   > A proxy's pod must have a `quilkin.dev/role` annotation set to the value `proxy` in order
-     for the management server to detect the pod as a proxy and push updates to it.
-
-   > Note that currently, the server can only discover resources within a single cluster.
-
-   ##### Admin server
-
-   In addition the gRPC server, a http server (configurable via `--admin-port`is also started to serve administrative functionality.
-   The following endpoints are provided:
-   - `/ready`: Readiness probe that returns a 5xx if communication with the Kubernetes api is problematic.
-   - `/live`: Liveness probe that always returns a 200 response.
-   - `/metrics`: Exposes Prometheus metrics.
+1. Cluster information is retrieved from [Agones] - the server watches for `Allocated`
+   [Agones GameServers] and exposes their IP address and Port as upstream endpoints to
+   any connected Quilkin proxies.
+
+   > Since an Agones GameServer can have multiple ports exposed, if multiple ports are in
+   > use, the server looks for the port named `default` and picks that as the endpoint's
+   > port (otherwise it picks the first port in the port list).
+
+1. Filter chain is configurable on a per-proxy basis. By default an empty filter chain is
```

Comment:
Since we're in here, shall we also document the usage of the annotation of `quilkin.dev/tokens` ?

https://github.com/googleforgames/quilkin/blob/b32e48af083795bb3e02c8915b34e3d8c5949623/xds/pkg/cluster/agones/agones.go#L138

Reply:
Added!